### PR TITLE
Fix: #17582 Tests to confirm the proper functionality of pimcore_placeholder with twig/twig 3.14

### LINKED ISSUE #17582: [Bug]: Outdated twig/twig dependency has a security vulnerability [CVE-2024-45411]
### Pimcore version

11.3.2

### Steps to reproduce

composer update

### Actual Behavior

`roave/security-advisories` and `trivy` security checks conflicts with `"twig/twig": "<1.44.8|>=2,<2.16.1|>=3,<3.14"` due to CVE-2024-45411:  https://www.cve.org/CVERecord?id=CVE-2024-45411

pimcore requires:
```
    "twig/extra-bundle": "^3.4.0",
    "twig/string-extra": "^3.3.5",
    "twig/twig": "^3.3.9",
```

### Expected Behavior

pimcore requires:
```
    "twig/twig": "^3.14",
```

### PR DESCRIPTION
**Pull request by michalananapps**
## Changes in this pull request  
Resolves #17582

## Additional info
Some tests have been added to confirm the proper functionality of pimcore_placeholder with twig/twig 3.14

### CODE REVIEW

**Comment by michalananapps** [comment]
recheck

**Comment by kingjia90** [comment]
@michalananapps Thank you for your PR!
The CLA should be rechecked after posting a comment with only this phrase 
`I have read the CLA Document and I hereby sign the CLA`

**Comment by michalananapps** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by markus-moser** [comment]
I can confirm that the current twig version seems to fix our problems with the output buffering. We still need to test all the cases in detail but from what I can see at the moment it looks good. 

Unfortunately the rybakit/twig-deferred-extension is still broken.

File: composer.json [hunk 1]
```diff
@@ -124,17 +124,16 @@
     "symfony/workflow": "^6.4",
     "symfony/yaml": "^6.4",
     "tijsverkoyen/css-to-inline-styles": "^2.2.3",
-    "twig/extra-bundle": "^3.4.0",
-    "twig/string-extra": "^3.3.5",
-    "twig/twig": "^3.3.9",
+    "twig/extra-bundle": "^3.13",
+    "twig/string-extra": "^3.13",
+    "twig/twig": "^3.14",
```
Comment (kingjia90) [inline composer.json]:
```suggestion
    "twig/extra-bundle": "^3.4.0",
    "twig/string-extra": "^3.3.5",
    "twig/twig": "^3.3.9",
```

File: composer.json [hunk 2]
```diff
@@ -124,17 +124,16 @@
...
     "umpirsky/country-list": "^2.0.6"
   },
   "conflict": {
     "pimcore/admin-ui-classic-bundle": "<1.5",
     "sabre/dav": "4.2.2",
     "symfony/symfony": "*",
-    "thecodingmachine/safe": "<2.0",
-    "twig/twig": ">=3.9.0"
+    "thecodingmachine/safe": "<2.0"
```
Comment (kingjia90) [inline composer.json]:
```suggestion
    "thecodingmachine/safe": "<2.0",
    "twig/twig": ">=3.9.0 <3.14.0"
```

**Review by kingjia90 — CHANGES_REQUESTED** [review]
Thank you for tests!
Overall LGTM, just the composer changes need to be reverted, because it has to be possible to install `twig` < `3.9.0`, in case some project need to be using the `twig-deferred-extension`.

**Pull request merged**